src/zenml/integrations/gcp/services/vertex_deployment.py

```diff
@@ -0,0 +1,431 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the Vertex AI Deployment service."""
+
+import re
+import time
+from typing import Any, Dict, Generator, List, Optional, Tuple
+
+from google.api_core import exceptions
+from google.cloud import aiplatform
+from google.cloud import logging as vertex_logging
+from pydantic import BaseModel, Field
+
+from zenml.client import Client
+from zenml.integrations.gcp.flavors.vertex_model_deployer_flavor import (
+    VertexBaseConfig,
+)
+from zenml.logger import get_logger
+from zenml.services import ServiceState, ServiceStatus, ServiceType
+from zenml.services.service import BaseDeploymentService, ServiceConfig
+
+logger = get_logger(__name__)
+
+# Increase timeout for long-running operations
+POLLING_TIMEOUT = (
+    1800  # Increased from 1200 to allow for longer deployment times
+)
+UUID_SLICE_LENGTH: int = 8
+
+
+def sanitize_vertex_label(value: str) -> str:
```

Comment:
Perfect, you already have a label sanitizer. You should use this in the model registry implementation too.